src/history/hs_rec.c

```diff
@@ -327,8 +326,13 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
     insert_cnt = 0;
     WT_TIME_WINDOW_INIT(&tw);
 
-    WT_RET(__wt_curhs_open(session, NULL, &hs_cursor));
-    F_SET(hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+    if (r->hs_cursor == NULL)
+        WT_RET(__wt_curhs_open(session, NULL, &r->hs_cursor));
+
+    cur_hs_read_committed = F_ISSET(r->hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
```

Comment:
This is fragile, as its possible someone in the future could set an entirely different flag on the cursor and change the behaviour in this function. We should save the flag state, set our flags, and return it to its original state.

Reply:
That's a good point, I made changes in 4a15e690cf132c43b8f4464a0d06498addbd0605